File: Examples/test-suite/lua/argcargvtest_runme.lua
require("import")	-- the import fn
import("argcargvtest")	-- import lib
v = argcargvtest

-- catch "undefined" global variables
local env = _ENV -- Lua 5.2
if not env then env = getfenv () end -- Lua 5.1
setmetatable(env, {__index=function (t,i) error("undefined global variable `"..i.."'",2) end})

largs = {"hi", "hola", "hello"}
assert(v.mainc(largs) == 3, "bad main typemap")

targs = {"hi", "hola"}
assert(v.mainv(targs, 0) == "hi", "bad main typemap")
assert(v.mainv(targs, 1) == "hola", "bad main typemap")
assert(v.mainv(targs, 2) == "<<NULL>>", "bad main typemap")

errorVal = 0
function try()
    mainv("hello", 1)
    errorVal = 1
end
assert(not pcall(try) and errorVal == 0, "bad main typemap")

v.initializeApp(largs)

-- Check that an empty array works.
empty_args = {}
assert(v.mainc(empty_args) == 0, "bad main typemap")
assert(v.mainv(empty_args, 0) == "<<NULL>>", "bad main typemap")

-- Check that empty strings are handled.
empty_string = {"hello", "", "world"}
assert(v.mainc(empty_string) == 3, "bad main typemap")
assert(v.mainv(empty_string, 0) == "hello", "bad main typemap")
assert(v.mainv(empty_string, 1) == "", "bad main typemap")
assert(v.mainv(empty_string, 2) == "world", "bad main typemap")
assert(v.mainv(empty_string, 3) == "<<NULL>>", "bad main typemap")

File: Examples/test-suite/php/argcargvtest_runme.php
<?php

require "tests.php";

// New functions
check::functions(array('mainc', 'mainv', 'initializeapp'));
// New classes
check::classes(array('argcargvtest'));
// No new vars
check::globals(array());

$largs = array('hi', 'hola', 'hello');
check::equal(mainc($largs), 3, 'Test main typemap 1');

$targs = array('hi', 'hola');
check::equal(mainv($targs, 0), 'hi', 'Test main typemap 2a');
check::equal(mainv($targs, 1), 'hola', 'Test main typemap 2b');
check::equal(mainv($targs, 2), '<<NULL>>', 'Test main typemap 2c');

$error = 0;
try {
    mainv('hello', 1);
    $error = 1;
}
catch (exception $e) {
}
check::equal($error, 0, 'Test main typemap 3');

initializeApp($largs);

# Check that an empty array works.
$empty_args = [];
check::equal(mainc($empty_args), 0, "test main typemap 4");
check::equal(mainv($empty_args, 0), '<<NULL>>', 'Test main typemap 4a');

# Check that empty strings are handled.
$empty_string = ["hello", "", "world"];
check::equal(mainc($empty_string), 3, "test main typemap 5");
check::equal(mainv($empty_string, 0), "hello", "test main typemap 6a");
check::equal(mainv($empty_string, 1), "", "test main typemap 6b");
check::equal(mainv($empty_string, 2), "world", "test main typemap 6c");
check::equal(mainv($empty_string, 3), "<<NULL>>", "test main typemap 6d");

check::done();
